floss/utils.py

```diff
@@ -42,6 +43,48 @@ def __call__(self, parser, namespace, values, option_string=None):
         setattr(namespace, self.dest, items)
 
 
+class InstallContextMenu(argparse.Action):
+    def __init__(self, option_strings, dest, nargs=None, **kwargs):
+        super(InstallContextMenu, self).__init__(option_strings, dest, nargs=0, **kwargs)
+
+    def __call__(self, parser, namespace, values, option_string=None):
+        # Theoretically, we don't need to check the platform again here,
+        # because non-Windows platforms will not accept the --install-right-click-menu parameter at all.
+        # This judgment is just to make the mypy type check pass.
+        # The same logic applies to `UninstallContextMenu` below.
+        if sys.platform == "win32":
+            import winreg as reg
+
+            menu_name = "Open with FLOSS"
+            menu_command = f'C:\\windows\\system32\\cmd.exe /K "python -m floss ^"%1^""'
+
+            shell_key = reg.OpenKey(reg.HKEY_CURRENT_USER, r"Software\\Classes\\*\\shell", 0, reg.KEY_SET_VALUE)
+            reg.SetValue(shell_key, menu_name, reg.REG_SZ, menu_name)
+
+            menu_key = reg.OpenKey(shell_key, menu_name, 0, reg.KEY_SET_VALUE)
+            reg.SetValueEx(menu_key, "AppliesTo", 0, reg.REG_SZ, "System.ItemName:exe")
```

Comment:
we can run FLOSS on any file, so we don't need this

Reply:
but ideally, we can set the icon to the FLOSS icon via the `Icon` value name

Reply:
It seems that we didn't include the icon for floss in the pip installation, so I only added the icon when the user uses a standalone exe.